framework/src/geomsearch/NearestNodeLocator.C

```diff
@@ -138,9 +138,25 @@ NearestNodeLocator::findNodes()
     const std::map<dof_id_type, std::vector<dof_id_type>> & node_to_elem_map =
         _mesh.nodeToElemMap();
 
+    // Convert trial master nodes to a vector of Points. This would be used to
+    // construct the Kdtree.
+    std::vector<Point> master_points(trial_master_nodes.size());
+    for (unsigned int i = 0; i < trial_master_nodes.size(); ++i)
+    {
+      const Node & node = _mesh.nodeRef(trial_master_nodes[i]);
+      for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
+        master_points[i](j) = node(j);
+    }
+
+    // Create object kd_tree of class KDTree using the coordinates of trial
+    // master nodes.
+    KDTree kd_tree(master_points);
+    kd_tree.buildTree();
```

Comment:
With the way you are constructing this object then calling build tree, are you sure you can't snag the dynamic memory you need all at once in the constructor?